src/components/Video.tsx

```diff
@@ -1,27 +1,39 @@
 import { FunctionComponent } from 'react'
 
-export const Video: FunctionComponent<{ filePath: string }> = ({ filePath }) => (
+interface Video {
+    source: {
+        webm: string
+        mp4: string
+    }
+    loop: boolean
+    caption: string // can also be used for alt
+    title: string
+}
+
+export const Video: FunctionComponent<Video> = ({ source, loop, caption, title }) => (
     <video
         className="w-100 h-auto shadow"
         width={1280}
         height={720}
         autoPlay={true}
         muted={true}
-        loop={true}
+        loop={loop}
         playsInline={true}
         controls={false}
+        title={title}
         // GCS does not set cookies, so we don't want Cookiebot to block this video based on consent
         data-cookieconsent="ignore"
     >
         <source
             type="video/webm"
-            src={`https://storage.googleapis.com/sourcegraph-assets/${filePath}.webm`}
+            src={`https://storage.googleapis.com/sourcegraph-assets/${source.webm}.webm`}
             data-cookieconsent="ignore"
         />
         <source
             type="video/mp4"
-            src={`https://storage.googleapis.com/sourcegraph-assets/${filePath}.mp4`}
+            src={`https://storage.googleapis.com/sourcegraph-assets/${source.mp4}.mp4`}
             data-cookieconsent="ignore"
         />
+        {caption}
     </video>
```

Comment:
For the use of `figcaption`, the the parent needs to be a `figure` element, even if the `figcaption` is optional

```suggestion
    <figure>
        <video
            className="w-100 h-auto shadow"
            width={1280}
            height={720}
            autoPlay={true}
            muted={true}
            loop={true}
            loop={loop}
            playsInline={true}
            controls={false}
            title={title}
            // GCS does not set cookies, so we don't want Cookiebot to block this video based on consent
            data-cookieconsent="ignore"
        >
            <source
                type="video/webm"
                src={`https://storage.googleapis.com/sourcegraph-assets/${filePath}.webm`}
                src={`https://storage.googleapis.com/sourcegraph-assets/${source.webm}.webm`}
                data-cookieconsent="ignore"
            />
            <source
                type="video/mp4"
                src={`https://storage.googleapis.com/sourcegraph-assets/${filePath}.mp4`}
                src={`https://storage.googleapis.com/sourcegraph-assets/${source.mp4}.mp4`}
                data-cookieconsent="ignore"
            />
            {caption && <figcaption>{caption}</figcaption>}
        </video>
    </figure>
```

Reply:
Thank you for catching that `<figcaption>` needs to be inside a `<figure>`! I agree with moving it into this component, and I removed the orphaned `<figcaption>`s from the blogpost markdown.

There are a few places where `Video` needs to appear without a caption while retaining the backup text, e.g., the Code Insights and Batch Changes pages. In this case, what do you think of differentiating between the two? Perhaps we could use a `showCaption: boolean` prop to display the `<figcaption>` in `<figure>`. This way, `caption` is still required and will always be embedded in the `<video>` element.

We can follow this pattern with our `YouTube` component and remove the caption in the markdown.

For example: [Video.tsx](https://github.com/sourcegraph/about/blob/59d0201eca1d496e535092f15d314bd3272f49d7/src/components/Video.tsx) and [YouTube.tsx](https://github.com/sourcegraph/about/blob/59d0201eca1d496e535092f15d314bd3272f49d7/src/components/YouTube.tsx)

```
         <source
            ...
            />
            {caption && caption}
        </video>
        {showCaption && <figcaption>{caption}</figcaption>}
    </figure>
```